src/dropdown/Dropdown.js

```diff
@@ -24,34 +22,46 @@ export default class Dropdown extends React.Component {
     this.logWarnings();
   }
 
-  getGeneralProps = () => omit(this.props, Object.keys(PropTypes));
-
   logWarnings = () => {
     if (this.props.children) {
       warn('You\'re passing children. Not expected behaviour');
     }
   };
 
-  getValueLinkProps = () => {
-    if (this.props.valueLink) {
-      return {
-        value: this.props.valueLink.value,
-        onChange: this.props.valueLink.requestChange
-      };
+  getValue = () => (
+    this.props.valueLink ? this.props.valueLink.value : this.props.value
+  );
+
+  valueToOption = (value, options) => {
+    if (t.String.is(value) || t.Number.is(value)) {
+      return find(options, { value });
     }
+    return value;
   };
 
-  getClassName = () => cx(this.props.className, themes[this.props.theme]);
-
-  render() {
-    // The order is important: props may override previous ones
-    return (
-      <Select
-        {...this.getGeneralProps()}
-        {...this.getValueLinkProps()}
-        {...this.getClassName()}
-      />
-    );
+  _onChange = () => this.props.valueLink ? this.props.valueLink.requestChange : this.props.onChange;
```

Comment:
1. `_onChange` is called with a `value`
2. this is an usual why of defining a `_onChange` function: why aren't you calling the prop `onChange` here?
I think you should do this:
```js
_onChange(value) => this.props.valueLink ? this.props.valueLink.requestChange(value) : this.props.onChange(value);
```

or if you prefer your version then please rename it to `getOnChange` (as that's what you're actually doing)
this.props.valueLink ? this.props.valueLink.requestChange() : this.props.onChange;